File: src/test/java/org/micoli/php/builders/BuiltinBuilder.java
package org.micoli.php.builders;

import org.micoli.php.tasks.configuration.runnableTask.Builtin;

public class BuiltinBuilder {
    private final Builtin shell;

    private BuiltinBuilder() {
        this.shell = new Builtin();
    }

    public static BuiltinBuilder create() {
        return new BuiltinBuilder();
    }

    public BuiltinBuilder withId(String id) {
        shell.id = id;
        return this;
    }

    public BuiltinBuilder withLabel(String label) {
        shell.label = label;
        return this;
    }

    public BuiltinBuilder withActionId(String actionId) {
        shell.actionId = actionId;
        return this;
    }

    public BuiltinBuilder withIcon(String icon) {
        shell.icon = icon;
        return this;
    }

    public Builtin build() {
        return shell;
    }
}

File: src/test/kotlin/org/micoli/php/builders/BuiltinBuilder.kt
package org.micoli.php.builders

import org.micoli.php.tasks.configuration.runnableTask.Builtin

class BuiltinBuilder private constructor() {
    private val shell: Builtin = Builtin()

    fun withId(id: String): BuiltinBuilder {
        shell.id = id
        return this
    }

    fun withLabel(label: String): BuiltinBuilder {
        shell.label = label
        return this
    }

    fun withActionId(actionId: String): BuiltinBuilder {
        shell.actionId = actionId
        return this
    }

    fun withIcon(icon: String): BuiltinBuilder {
        shell.icon = icon
        return this
    }

    fun build(): Builtin = shell

    companion object {
        fun create(): BuiltinBuilder = BuiltinBuilder()
    }
}
